tests/test_config_command.rs

```diff
@@ -194,13 +217,18 @@ fn test_config_layer_override_env() {
     .unwrap();
     let stdout = test_env.jj_cmd_success(&repo_path, &["config", "list", config_key]);
     insta::assert_snapshot!(stdout, @r###"
+    ui.editor="env-base"
+    ui.editor="user"
     ui.editor="repo"
     "###);
 
     // Environment override
     test_env.add_env_var("JJ_EDITOR", "env-override");
     let stdout = test_env.jj_cmd_success(&repo_path, &["config", "list", config_key]);
     insta::assert_snapshot!(stdout, @r###"
+    ui.editor="env-base"
+    ui.editor="user"
+    ui.editor="repo"
```

Comment:
> Also list all values for a given config key name (ordered by precedence)

What's the main point of this change? I think printing overridden values is rather confusing.

Reply:
That's how `git config -l` works. Since the command is called "list" it makes sense to me that it would report multiple (and not consider it an "error" if there are zero). Also if we're including options like `--show-origin` it's helpful to include that original context.

We did talk about whether it makes sense to have a "get" vs "list" command a while back, and I _would_ expect a "get" command to report only the final value. I'd also envisioned colorizing this output eventually and shading overridden values a muted gray, plus maybe an option for whether to list overridden values.

Reply:
I considered it a "list" matching the given name or pattern, not a list of values including overridden ones. Maybe I'm too familiar with hg than git.

Colorizing output isn't always doable, so it might be better to comment out overridden values. It also conforms to TOML where duplicated keys are invalid.
```
# ui.editor="env-base"
# ui.editor="user"
ui.editor="repo"
```
(But I still prefer not showing overridden values by default.)

Reply:
Ooh, I like that idea commenting out.

FWIW another analog is Chrome Devtools will list all CSS rules with a strikethrough for overridden ones.

Anyway let's get some input from @martinvonz and others. I could imagine putting those all behind an `--include-overridden` option, but that could also be saved as an improvement to add as follow-up.

Reply:
I also like the idea of commenting them out so it's valid TOML, but also so it's clear that they're not in effect. I think we should still comment it out if the user did `jj config list --user ui.editor` in the example above.

I think we should still default to not including overridden values. The option to see them seems useful for troubleshooting, but I feel like I've almost always wanted to just see the applied config when I've used `hg config` in the past. Do you know why git includes them by default, or do you have a usecase for it?

Reply:
Alright, I can just exclude them for now and work on an `--include-overridden` option later (showing them commented-out).

Still thinking nonexistent names should be treated as overall "success" but with a hint to stderr, SG?

And one other wrinkle I thought of: I'd kind of expect built-in default values to not show *unless* you'd specifically requested those names. E.g. if I didn't have any explicit `colors` values configured their defaulted values wouldn't show up in `config list` but maybe should in `config list colors`. Opinions on that?

Reply:
Sounds good.

I would probably not include built-in default values without a `--include-defaults` flag, especially since it depends so much on exactly how we've implemented the defaults - some defaults are applied to the `LayeredConfigs`, and some are applied at the use site. Treating `config list colors` specially seems like it would lead to confusion.

Reply:
Okay, the good news is I got the behavior of hiding overridden values implemented. The bad news is my code for doing it looks... just terrible. (Although that's also potentially good news for me, because I'm probably going to learn a lot getting suggestions from more experienced rustaceans about how to do it properly 😁).

Please give me some suggestions to fix this mess. I'll leave a few inline comments about parts where I got stuck...